### ISSUE TITLE
Question about progressive growing

### ISSUE THREAD

**Issue by mrgloom** [rosinality/style-based-gan-pytorch#62]
> 8/16/32/64/128/256/512/1024

Will results for each pyramid scale will be good? or it only guaranteed that result for highest resolution will be good?

**Comment by kobrien-Novetta** [comment]
In my experience, results from all of the resolutions should be good, as
the model progresses up. It is unlikely that the final resolution will be
good if the earlier resolutions are bad .

**Comment by rosinality** [comment]
Samples will be good, but it isn't very high quality as training iterations would be somewhat short at low resolution phases.

**Comment by mrgloom** [comment]
I was saving images each 10k steps and this is an image before 64p stage, i.e. it's 32p images grid.
And images obviously have artifacts.
![110000](https://user-images.githubusercontent.com/4003908/68542460-c6f70080-03bd-11ea-87d0-33fc79b08a4d.png)
120k - 64p images grid.
![120000](https://user-images.githubusercontent.com/4003908/68542497-45ec3900-03be-11ea-8ae2-4050ae30ea26.png)

Is this expected, i.e. we will have 'near perfect quality' only at final resolution in progressive growing setting? Is it meaninglessly to have 'near perfect quality' at each intermediate resolution?
i.e. something like this https://github.com/akanimax/BMSG-GAN/

**Comment by rosinality** [comment]
Yes, it will be not very high quality especially at lower resolutions. And you will need truncation tricks for better samples. You don't need to have high quality samples at lower resolutions for the final quality, but if you need it, I think you can train more steps at each resolutions.

**Comment by mrgloom** [comment]
For now I have trained model for 320k+ steps in ~130 hours.

320k - 128p images grid:
![320000](https://user-images.githubusercontent.com/4003908/68895589-ab338780-073a-11ea-86c3-b58592a7fdfa.png)

For now it's training at 128p resolution (maximum resolution in my setting) with alpha=1.0
As we can see quality of images still not perfect, does it need to train more? How 'good' images should look before truncation trick?

**Comment by rosinality** [comment]
I think it is almost enough if you use truncation tricks.
